app/scripts/modules/core/src/presentation/forms/README.md

```diff
@@ -1,130 +1,230 @@
+The `FormField` component accepts "all the props", organizes them,
+and passes them down to the Input and Layout in the correct spots.
+
+The props accepted by FormField are:
 
-### Example (providing custom Layout component)
+- `input`: the Input component to use
+- `layout`: the Layout component to use. (optional, `StandardFieldLayout` is used by default)
+- `label`, `help`, `required`, `actions` (see `IFieldLayoutPropsWithoutInput`)
+- `name`, `value`, `onChange`, `onBlur`, (see `IControlledInputProps`)
+- `touched`, `error` (see: `IValidationProps`)
+
+### Custom Input Example:
+
+This example uses an input which is defined inline:
 
 ```js
-<TextField
-  value={this.state.timelineName}
-  onChange={timelineName => this.setState({ timelineName })}
-  label="Timeline Name"
-  FieldLayout={MyFancyFieldLayoutComponent}
+<FormField
+  name="email"
+  label="email address"
+  input={({ field, validation }) => <input type="text" className={!!validation.error && 'error'} {...field} />}
+  value={this.state.email}
+  onChange={evt => this.setState({ email: evt.target.value })}
+  error={this.state.email.indexOf('@') === -1 && <span>Please enter a valid email</span>}
 />
 ```
 
-### Example with an error message:
+### Layout Example:
+
+This example uses a custom layout to render the error above the input
 
 ```js
-<TextField
-  value={this.state.timelineName}
-  onChange={timelineName => this.setState({ timelineName })}
-  label="Timeline Name"
-  error={this.state.error && <div>Error: {this.state.error}</div>}
+<FormField
+  name="email"
+  label="email address"
+  input={TextInput}
+  layout={({ error, input }) => (
+    <div>
+      <div style={{ background: 'red' }}>{error}</div>
+      <div>{input}</div>
+    </div>
+  )}
+  value={this.state.email}
+  onChange={evt => this.setState({ email: evt.target.value })}
+  error={this.state.email.indexOf('@') === -1 && <span>Please enter a valid email</span>}
 />
 ```
 
-<img width="814" alt="screen shot 2018-06-09 at 3 44 19 pm" src="https://user-images.githubusercontent.com/2053478/41196697-34723ad6-6bfc-11e8-90da-8b46973c2461.png">
+## Inputs
 
-### Example using Formik
+An Input is a react component which is rendered by a `FormField` or `FormikFormField` and:
 
-Field components wrap a controlled input component.
-However, a Formik wrapper is added to Field components, as a static `Formik` property.
+- accepts user input
+- emits change events when the input changes
+- visually indicates an invalid validation state
+
+An Input receives two important props (see `IFormInputProps`):
+
+- `field`: an `IControlledInputProps` object which provides convenient props for rendering controlled form inputs.
+  - `name`, `value`, `onChange`, `onBlur`
+  - The contents of this prop is typically spread onto an input, e.g., `<input {...field} />`
+- `validation`: The current validation state of the input used to visually indicate the current validation state of the input
```

Comment:
department of redundancy department

Reply:
This office will not tolerate redundancy in this office